util/Directories.h

```diff
@@ -111,5 +111,15 @@ FO_COMMON_API boost::filesystem::path GetPath(const std::string& path_string);
 /** Returns if path exists and is a regular file */
 FO_COMMON_API bool IsExistingFile(const boost::filesystem::path& path);
 
+/** All paths contained in a directory filtered by a functor
+ * 
+ * @param[in] abs_dir_path Absolute path to directory
+ * @param[in] pred Predicate functor accepting a boost::filesystem::path constant reference
+ * @param[in] recursive_search If true, recurses into sub-directories
+ *
+ * @return vector List of filesytem objects found in @p abs_dir_path which satisfy @p pred */
+FO_COMMON_API std::vector<boost::filesystem::path> PathsInDir(const boost::filesystem::path& abs_dir_path,
```

Comment:
Eventually, I hope/plan to replace `ListDir` with [one of three forms](https://github.com/freeorion/freeorion/compare/master...dbenage-cx:common_file_op#diff-4df2a52c1d1829d7c478cd4d738cbe24R121), along with other such functions, removing use of boost/filesystem/operations.hpp in most other files.
(branch is WIP, and excludes inclusion in larger codebase for clarity.)

If I do not find time to resolve that branch soon, will do a refactor pass to utilize `PathsInDir` and `IsExistingFile` more thoroughly.

Reply:
Very well.

A feature that I would like to see included if possible is to prevent access to paths outside of the directories provided by the user: resource directory, saved games directory etc.  All paths should be relative to and not follow links out of directories that the user has allowed FreeOrion to use.

Reply:
👍 Agree, designed with such purpose in mind, but not necessarily restricting all access just yet. Will try to follow up with forum post.